agora/Agora/Effect/TreasuryWithdrawal.hs

```diff
@@ -230,10 +239,11 @@ treasuryWithdrawalValidator = plam $
 
       pure . popaque $ pconstant ()
   where
-    -- Make sure that all the receivers get the correct payment and return the
+    -- Make sure that all the receivers get the correct payment, return the
     --  remaining outputs.
+    --
+    -- This function is not hoisted cause it's used only once.
```

Comment:
While this is probably fine, there's no reason to avoid hoisting here, because if a hoisted term is used only once, it'll get inlined anyway IIRC.